code/datums/components/multitile.dm

```diff
@@ -0,0 +1,54 @@
+///Lets multitile objects have dense walls around them based on the coordinate map
+/datum/component/multitile
+	///Reference to all fillers
+	var/list/all_fillers = list()
+
+/*
+  * These should all be done in this style. It represents a coordinate map of the grid around `src`.
+  * The src itself should always have no density, as the density should be set on the atom and not with a filler
+  * list(
+		list(0, 0, 0,		   0, 0),
+		list(0, 0, 0,		   0, 0),
+		list(0, 0, MACH_CENTER, 0, 0),
+		list(0, 0, 0,		   0, 0),
+		list(0, 0, 0,		   0, 0)
+	)
+ */
+
+//distance_from_center does not include src itself
+/datum/component/multitile/Initialize(distance_from_center, new_filler_map)
```

Comment:
I don't really see why we need to specify `distance_from_center` as an argument when we should already have that information in our provided map. 

I think you could probably parse this map a little bit to get some more information on it/to provide some sanity checks. You can make sure all the lists are the same length, that MACH_CENTER is indeed in the center (and somewhere in the lists, for that matter), and then determine the distance from center from there.

Reply:
`MACH_CENTER` is just a define for readability, it's just `0`, so no way to really check if it exists. The rest I can do

Reply:
What I mean is that you could make it like -1 or something, and then add some special logic to account for it while processing.

Reply:
Discussed in discord